### DISCUSSION TITLE
Hooking into transform event Classic Block -> Any gutenberg block

### DISCUSSION THREAD

**Discussion by furai** [WordPress/gutenberg#63670] [Q&A]
Hey,

I'm looking for a way to get the event when Classic Block gets transformed. I want to alter what it is transformed to. Are there any hooks for that? What should I be looking for?

Maybe I'm thinking about it wrong and I should overwrite some built-in functions for transformations if such exist? 

What I'm after is when transforming let's say classic table to gutenberg editor table - I want to preserve custom classes added on this table. And want to move them into table advanced tab into additional classes option. Is it even possible to do?

Any help highly appreciated. I've been stuck on this for a longer while now but I've learnt quite a bunch in the process.

Cheers,
Furai

**Comment by Marc-pi** [comment]
maybe there is an answer through this plugin's code https://wordpress.org/plugins/convert-to-blocks/ (I did not investigate)

**Reply by furai** [reply]
So the usual answers I'm getting with Gutenberg. Instead of getting answers like "here's documentation page just about that or something similar, take a look!" I'm getting answers with links to various blogs on the Internet or I'm told to check the code. 

Thank you, nonetheless. I'll check this plug-in and its code out!

EDIT: I did look through the code and it's not what I'm after.

**Comment by sntran** [comment]
I had to do something similar but could not find the answer. I ended up hooking to "blocks.getSaveElement" filter to parse and change the content of `core/reform` block:

```js
wp.hooks.addFilter(
  "blocks.getSaveElement",
  "core/freeform/cleanup",
  function(element, blockType, attributes) {
    if (!element) {
      return;
    }

    if (blockType.name !== "core/freeform") {
      return element;
    }

    const doc = new DOMParser().parseFromString(attributes.content, "text/html");
    for (const paragraph of doc.querySelectorAll("p")) {
       // Modify paragraph...
    }

    element.props.children = doc.body.innerHTML;

    return element;
  }
);
```

The idea is to use the "Convert to block" button on the Classic block, which would apply the "blocks.getSaveElement". The resulting element from the hook would then be converted to Gutenberg blocks.

**Comment by t-hamano** [comment]
> I'm looking for a way to get the event when Classic Block gets transformed.

[Here](https://github.com/WordPress/gutenberg/blob/47dd855997134d05c4c985fa9a10ecfcc8a58556/packages/block-library/src/freeform/convert-to-blocks-button.js#L22-L25) is the event that occurs when you press the "Convert to blocks" button. You might be able to hack this event, but it would probably be quite complicated, as the processing is hard-coded.

> I want to preserve custom classes added on this table. And want to move them into table advanced tab into additional classes option. Is it even possible to do?

I am unable to reproduce this issue. What HTML did you use? I tested with the following HTML:

```html
<h2 class="my-heading">Heading</h2>
<p class="my-paragraph">Paragraph</p>

<ul class="my-list">
 	<li>List Item</li>
 	<li>List ITem</li>
</ul>
<table class="my-table">
<tbody>
<tr>
<td>Table Cell</td>
</tr>
</tbody>
</table>
```

https://github.com/user-attachments/assets/73b98750-df02-41a8-bbd2-e9c67e4c8844
